src/components/layout/sync-button.tsx

```diff
@@ -20,61 +21,49 @@ const SyncButton = () => {
   const [syncing, setSyncing] = useState(false);
   const [syncSuccess, setSyncSuccess] = useState(false);
   const [syncHandlerOpen, setSyncHandlerOpen] = useState(false);
+  const [canSync, setCanSync] = useState(false);
   const [dailyEntries, setDailyEntries] = useState<DailyEntry[]>();
-  const [persons, setPersons] = useState<Person[]>();
-  const { personsApi, dailyEntriesApi } = useApi();
+  const { dailyEntriesApi } = useApi();
   const userProfile = useAtomValue(userProfileAtom);
-
-  useEffect(() => {
-    fetchPersons();
-    fetchDailyEntries();
-  }, []);
-
-  /**
-   * fetching persons
-   */
-  const fetchPersons = async () => {
-    try {
-      const tempPersons = await personsApi.listPersons({});
-      setPersons(tempPersons);
-    } catch (error) {
-      setError(`${strings.error.fetchFailedGeneral}, ${error}`);
-    }
-  };
+  const persons = useAtomValue(personsAtom);
+  const loggedInPerson = persons?.find((person: Person) => person.keycloakId === userProfile?.id);
 
   /**
    * fetching daily entries
    */
   const fetchDailyEntries = async () => {
     try {
-      const loggedInPerson = persons?.find(
-        (person: Person) => person.keycloakId === userProfile?.id
-      );
       const fetchedDailyEntries = await dailyEntriesApi.listDailyEntries({
-        personId: loggedInPerson?.id || config.person.forecastUserIdOverride
+        personId: loggedInPerson?.id || config.person.forecastUserIdOverride,
+        vacation: false
       });
       setDailyEntries(fetchedDailyEntries);
     } catch (error) {
       setError(`${strings.error.fetchFailedNoEntriesGeneral}, ${error}`);
     }
+    setCanSync(true);
   };
 
+  useEffect(() => {
+    if(persons.length)
+      {
+        fetchDailyEntries();
+      }
+  }, [persons]);
+
   /**
    * Update latest daily entry date
    */
   const updateSyncFromDailyEntries = async () => {
     let dailyEntryDates: DateTime[] = [];
+    try {
 
-    if (dailyEntries?.length) {
-      dailyEntryDates = dailyEntries.map((dailyEntry) => DateTime.fromJSDate(dailyEntry.date));
-    } else {
-      const fetchedDailyEntries = dailyEntries;
-
-      if (fetchedDailyEntries?.length) {
-        dailyEntryDates = fetchedDailyEntries.map((dailyEntry) =>
-          DateTime.fromJSDate(dailyEntry.date)
-        );
+      if (dailyEntries?.length) {
+        dailyEntryDates = dailyEntries.map((dailyEntry) => DateTime.fromJSDate(dailyEntry.date));
       }
+    } catch (error) {
+      console.error("Error fetching daily entries:", error);
```

Comment:
We generally don't use console.error in the project as we have the setError which handles errros. So I think this can be removed, if not it should be localized